static/js/lib/api.js

```diff
@@ -243,15 +244,20 @@ export function addCourseEnrollment(courseId: string) {
 }
 
 export function getCoupons(): Promise<Coupons> {
-  return fetchJSONWithCSRF('/api/v0/coupons/');
+  return fetchJSONWithCSRF('/api/v0/coupons/').then(coupon => (
+    // turn `amount` from string into decimal
+    R.evolve({amount: Decimal}, coupon)
```

Comment:
This should be an array, not an object:

![screenshot from 2017-01-26 10-45-44](https://cloud.githubusercontent.com/assets/863262/22338189/c0e5c478-e3b4-11e6-8be6-7a6b65fda69a.png)